vortex-duckdb-ext/src/convert/dtype.rs

```
@@ -76,12 +129,22 @@ impl TryFrom<&DType> for LogicalType {
 
                 return LogicalType::struct_type(child_types, child_names);
             }
-            // TODO: Add support for Decimal, Extension, List
-            _ => {
-                return Err(vortex_err!(
-                    "Unsupported DType for DuckDB conversion: {:?}",
-                    dtype
-                ));
+            DType::Decimal(decimal_dtype, _) => {
+                return LogicalType::decimal_type(
+                    decimal_dtype.precision(),
+                    decimal_dtype.scale() as _,
```

We cannot do this, you can negative scales

---

Can just fail loudly for now